From 3770d5fee62eb09dba5a71ccf178cc6856c7f1b6 Mon Sep 17 00:00:00 2001
From: Sami Liedes <sami.liedes@iki.fi>
Date: Sun, 19 Jan 2014 06:33:55 +0200
Subject: [PATCH 1/6] STP parsers: Fix build on Debian unstable

Recent versions of Bison have dropped the long-deprecated support for
YYPARSE_PARAM, so convert the parsers to use %parse-param instead.

Adapted from
https://github.com/delcypher/stp/commit/ece1a55fb367bd905078baca38476e35b4df06c3

Signed-off-by: Sami Liedes <sami.liedes@iki.fi>
---
 stp/src/parser/cvc.y  | 4 +++-
 stp/src/parser/smt.y  | 4 +++-
 stp/src/parser/smt2.y | 5 ++++-
 3 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/stp/src/parser/cvc.y b/stp/src/parser/cvc.y
index 41a8ee1..03e8ff1 100644
--- a/stp/src/parser/cvc.y
+++ b/stp/src/parser/cvc.y
@@ -22,7 +22,6 @@
 #define YYMAXDEPTH 1048576000
 #define YYERROR_VERBOSE 1
 #define YY_EXIT_FAILURE -1
-#define YYPARSE_PARAM AssertsQuery
   
   extern int cvclex(void);
   extern char* yytext;
@@ -32,9 +31,12 @@
     FatalError("");
     return YY_EXIT_FAILURE;
   };
+  int yyerror(void* AssertsQuery, const char* s) { return yyerror(s); }
   
   %}
 
+%parse-param {void* AssertsQuery}
+
 %union {
 
   unsigned int uintval;                 /* for numerals in types. */
diff --git a/stp/src/parser/smt.y b/stp/src/parser/smt.y
index 1e056db..7032743 100644
--- a/stp/src/parser/smt.y
+++ b/stp/src/parser/smt.y
@@ -54,15 +54,17 @@
     FatalError("");
     return 1;
   }
+  int yyerror(void* AssertsQuery, const char* s) { return yyerror(s); }
 
   ASTNode query;
 #define YYLTYPE_IS_TRIVIAL 1
 #define YYMAXDEPTH 104857600
 #define YYERROR_VERBOSE 1
 #define YY_EXIT_FAILURE -1
-#define YYPARSE_PARAM AssertsQuery
   %}
 
+%parse-param {void* AssertsQuery}
+
 %union {  
   // FIXME: Why is this not an UNSIGNED int?
   int uintval;                  /* for numerals in types. */
diff --git a/stp/src/parser/smt2.y b/stp/src/parser/smt2.y
index 8c6f465..35d96b2 100644
--- a/stp/src/parser/smt2.y
+++ b/stp/src/parser/smt2.y
@@ -65,6 +65,8 @@
     return 1;
   }
 
+  int yyerror(void *AssertsQuery, const char *s) { yyerror(s); }
+
   ASTNode querysmt2;
   vector<ASTVec> assertionsSMT2;
     
@@ -72,9 +74,10 @@
 #define YYMAXDEPTH 104857600
 #define YYERROR_VERBOSE 1
 #define YY_EXIT_FAILURE -1
-#define YYPARSE_PARAM AssertsQuery
   %}
 
+%parse-param {void *AssertsQuery}
+
 %union {  
   unsigned uintval;                  /* for numerals in types. */
   //ASTNode,ASTVec
-- 
1.9.0

